### ISSUE TITLE
error Valid CSRF Token Required

### ISSUE THREAD

**Issue by ParmaU** [osTicket/osTicket#4262]
Dear Team,

The system information as follow:

    64bit Linux Mint 18.1 Serena 
    Xampp ver 7.2.1
    PHP 7.2.1
    osTicket version 1.10.2
	ServerDB  10.1.30-MariaDB
	
	after updating the system from 1.10.1 to authentication it tells me
	Valid CSRF Token Required

**Comment by ntozier** [comment]
Well for starters osTicket does not support PHP 7.1+ at this time.

This also appears to be a duplicate of https://github.com/osTicket/osTicket/issues/3546 and https://github.com/osTicket/osTicket/issues/3256

**Comment by jorgemvc** [comment]
@ParmaU 

Just see fix #3546 

All you have to do is edit the file class.ostsession.php and insert this line at 193 row 
`$this->data->session_data = "";`

After the edition
`191 catch (DoesNotExist $e) {`
 `192    $this->data = new SessionData(['session_id' => $id]);`
 `193    $this->data->session_data = "";`

That worked for me

**Comment by ParmaU** [comment]
thank you, done and it works

**Comment by FreddyRichard** [comment]
Thanks @jorgemvc, your help solved the problem. The support is appreciated. 

I did this on a mac (macOS High Sierra 10.13.6) with PHP 7.1.21.

Thank you,
Greetings.

**Comment by Horus2Anubis** [comment]
Thank you very much!!! @jorgemvc, your help me solve the problem. The support is very appreciated.
I did this on a Windows 10 IIS  with PHP 7.2.7
Thank you, and have a great new year!!!
Greetings.

**Comment by goff2512** [comment]
Thanks! Worked for me too.  I appreciate it!

**Comment by memetoec** [comment]
I'm running my instance of OSticket on a web hotell, they run PHP Version 7.2.14, the fix @jorgemvc proposed worked like a charm for my OSticket install on PHP  7.2.14.

**Comment by steve-huynh** [comment]
@jorgemvc , Thanks for your advice. It works for mine, too.

**Comment by dfstab** [comment]
osTicket (v1.10.5)

Perfect one liner! I wonder why that is still an issue in Q2/2019?

**Comment by JediKev** [comment]
@dfstab 

Because we addressed this in `v1.11.x`, not `v1.10.x`.

Cheers.

**Comment by jswails** [comment]
this fix still working great on php 7.2.19 with latest osticket. thanks for that!

**Comment by aneurinprice** [comment]
I'm seeing this in an ubuntu container with php7.2.19.

**Comment by jswails** [comment]
There is a fix for that on the forum. You have to do a workaround. Its not
too difficult as I just did it a few days ago.

**Comment by aneurinprice** [comment]
@jswails Is this the fix you refer to?

> All you have to do is edit the file class.ostsession.php and insert this line at 193 row
> $this->data->session_data = "";

If so, I have tried this and no joy. You can see for yourself: https://gitlab.nyeprice.space/moby/osticket-docker

**Comment by jswails** [comment]
Yes thats the one that worked for me on ubuntu server 18.04 and php 7.2.
Thats unfortunate that didnt work for you. Hmm.

**Comment by DePorto** [comment]
Thank you, that trick adding the 193 line worked for me.

**Comment by hazemmatouk** [comment]
I fixed this issue by moving session_start() to first line of constractor
include/class.ostsession.php
